### PR TITLE
EC samples for verifying a singature, + serialization

### PR DESCRIPTION
**Pull request by ralienpp**
This extends the samples in the context of #3030.

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, ok to test

**Comment by ralienpp** [comment]
Aha, I see that it attempts to run the sample and there's something wrong with that. I will investigate.

```
Warning, treated as error:
WARNING: **********************************************************************
File "hazmat/primitives/asymmetric/ec.rst", line 75, in default
Failed example:
    verifier.verify()
Expected nothing
Got:
    True
```

**Comment by ralienpp** [comment]
I applied a few  changes to address the issues Jenkins found in the docs, but at this point it fails in a place that is outside of my reach:

```
ERROR: InvocationError: '/var/lib/jenkins/jobs/cryptography-pr-docs/workspace/cryptography/.tox/docs/bin/sphinx-build -W -b spelling docs docs/_build/html'
```

**Comment by reaperhulk** [comment]
The output is a bit confusing, but further up you can see it says:

```hazmat/primitives/asymmetric/ec.rst:594:Deserialization:```

which is what's causing the error. Namely, it doesn't think "Deserialization" is a word. You can add it to the spelling list :) (`docs/spelling_wordlist.txt`)

**Comment by ralienpp** [comment]
The machines appear to be happy now. As for the contents of the doc itself, let me know what can be improved in it.

File: docs/hazmat/primitives/asymmetric/ec.rst [hunk 1]
```diff
@@ -61,6 +61,21 @@ Elliptic Curve Signature Algorithms
     :func:`~cryptography.hazmat.primitives.asymmetric.utils.decode_dss_signature`.
 
 
+    Verification requires the public key, the signature itself, the signed data and knowledge of the hashing algorithm that was used when producing the signature:
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/ec.rst]: Nit: could you put a comma after `the signed data`. We generally use Oxford comma in our docs.

File: docs/hazmat/primitives/asymmetric/ec.rst [hunk 2]
```diff
@@ -61,6 +61,21 @@ Elliptic Curve Signature Algorithms
...
+    >>> public_key = private_key.public_key()
+    >>> verifier = public_key.verifier(signature, ec.ECDSA(hashes.SHA256()))
+    >>> verifier.update(b"this is some data I'd like")
+    >>> verifier.update(b" to sign")
+    >>> verifier.verify()
+    True
+
+    The last call will either return ``True`` or raise an :class:`~cryptography.exceptions.InvalidSignature` exception.
+
+    .. NOTE::
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/ec.rst]: this should be lowercase

File: docs/hazmat/primitives/asymmetric/ec.rst [hunk 3]
```diff
@@ -533,6 +548,70 @@ Key Interfaces
...
+    ...     encoding=serialization.Encoding.PEM,
+    ...     format=serialization.PublicFormat.SubjectPublicKeyInfo
+    ...     )
+    >>> serialized_public.splitlines()[0]
+    '-----BEGIN PUBLIC KEY-----'
+
+This is the part that you would normally share with the rest of the world.
+
+
+Deserialization
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/ec.rst]: In the `RSA` section we call this "Key loading" (and serialization is called "Key serialization"). Consistency is probably a good idea (although DSA doesn't match either of them so (¯\_(ツ)_/¯)

**Comment by ralienpp** [comment]
I applied the changes.

**Comment by reaperhulk** [comment]
@ralienpp Thanks again for the PR, feel free to send a PR adding yourself to `AUTHORS.rst` if you haven't already!

**Comment by ralienpp** [comment]
Great, thanks :-)

Can you also have a look at issue 2158?

**Pull request merged**